File: app/src/main/java/de/gymnasium_lappersdorf/gymlapapp/MainActivity.java
package de.gymnasium_lappersdorf.gymlapapp;

import android.support.design.widget.NavigationView;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;

import de.gymnasium_lappersdorf.gymlapapp.Navigation.NavDrawerOnclickHandler;
import de.gymnasium_lappersdorf.gymlapapp.Stundenplan.DatabaseHandler;

public class MainActivity extends AppCompatActivity {

    DrawerLayout drl;
    Toolbar tb;
    NavDrawerOnclickHandler handler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //initializing Stundenplan-DB at app start
        DatabaseHandler.INSTANCE.initialize(this);

        //setting Toolbar instead of Actionbar
        tb = findViewById(R.id.toolbar_main);
        setSupportActionBar(tb);

        //navigation Drawer
        drl = findViewById(R.id.drawer_layout_main);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(MainActivity.this, drl, tb, R.string.drawer_close, R.string.drawer_open);
        drl.addDrawerListener(toggle);
        try {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        } catch (NullPointerException e) {
            e.printStackTrace();
        }
        getSupportActionBar().setHomeButtonEnabled(true);
        toggle.syncState();
        NavigationView nv = findViewById(R.id.navigation);

        handler = new NavDrawerOnclickHandler(MainActivity.this);
        nv.setNavigationItemSelectedListener(handler);
        //setting HomeFragment as default fragment when launching Activity
        handler.setHomeFragment();
    }


}

File: app/src/main/java/de/gymnasium_lappersdorf/gymlapapp/MainActivity.kt
package de.gymnasium_lappersdorf.gymlapapp

import android.os.Bundle
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import de.gymnasium_lappersdorf.gymlapapp.Navigation.NavDrawerOnClickHandler
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //setting Toolbar
        setSupportActionBar(toolbar_main)

        //setting up Navigation-Drawer
        val toggle = ActionBarDrawerToggle(
                this,
                drawer_layout_main,
                toolbar_main,
                R.string.drawer_close,
                R.string.drawer_open
        )
        drawer_layout_main.addDrawerListener(toggle)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeButtonEnabled(true)
        toggle.syncState()
        val handler = NavDrawerOnClickHandler(this)
        navigation.setNavigationItemSelectedListener(handler)
    }
}